src/utils/crypto_utils/inc/key_data.h

```diff
@@ -0,0 +1,23 @@
+/**
+ * @file key_data.h
+ * @brief Declares the KeyData structure for the encryption/decryption functions
+ *
+ * @copyright Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ */
+
+#include <azure_c_shared_utility/constbuffer.h>
+
+#ifndef KEY_DATA_H
+#define KEY_DATA_H
+
+/***
+ * @brief Opaque type for callers to transfer KeyData, has the ability to add other keys and information depending on the input
```

Comment:
nit: it's not opaque -- the fields are available.  an opaque type would be

`typedef void * KeyData; // opaque type. `